protos/sources/data-node/api/v2/trading_data.proto

```diff
@@ -475,9 +475,22 @@ service TradingDataService {
 
   // List liquidity provisions
   //
-  // Get a list of liquidity provisions for a given market.
+  // Get a list of liquidity provisions for a given market
+  // DEPRECATED: When a liquidity provider amends a provision, and it's accepted by the network, the pending
+  // provision is returned by the API instead of the provision that is currently active.
+  // Use ListLiquidityProvisionsWithPending instead.
   rpc ListLiquidityProvisions(ListLiquidityProvisionsRequest) returns (ListLiquidityProvisionsResponse) {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Liquidity"};
+    option deprecated = true;
+  }
+
+  // List liquidity provisions
+  //
+  // Get a list of liquidity provisions for a given market. This is similar to the list liquidity provisions API
```

Comment:
If the old API is being deprecated then we should treat it such that it no longer exists anymore, so we shouldn't really reference in the doc string here.

Also the `WithPending` bit feels strange? If the old API is deprecated and should no longer be used (and maybe even eventually removed completely) we're left with this one API called `ListLiquidityProvisionsWithPending` which just reads weird given there isn't a `ListLiquidityProvisionsWithoutPending`.

Can we not call is `ListAllLiquidityProvisions`? It will then at least keep in line with the name we used when we deprecated `ListPositions` in favour of `ListAllPositions`?

Reply:
Yeah, I couldn't think of a better name, I didn't like the ListAll... for the API name but you make a good point in that we have done this in the past and @candida-d also suggested that so let's go with it.

I mention the deprecated API in the doc string because it's only deprecated and it still exists, and if/when we remove the old API, we can update the documentation, but I see your point and will remove it.